distributed/client.py

```diff
@@ -807,6 +820,120 @@ class VersionsDict(TypedDict):
     client: dict[str, dict[str, Any]]
 
 
+_T_LowLevelGraph: TypeAlias = dict[Key, tuple]
+
+
+def is_nested(iterable):
+    for item in iterable:
+        if (
+            isinstance(item, Iterable)
+            and not isinstance(item, str)
+            and not isinstance(item, bytes)
+        ):
+            return True
+    return False
+
+
+class MapLayer(Layer):
```

Comment:
I think I'd prefer a leading underscore as well. I don't want users to actually interact with this thing